src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final byte RESIZE_CAPACITY = 2;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeTable();
+        Node<K, V> newNode = new Node<>(key, value);
```

Comment:
we construct a new node without knowing if we need it. Let's not.

Reply:
we need this because we need to work with that

Reply:
Don't need it if we change the value in the existing node (first 'if' block)
We have a potentially redundant constructor call and a memory loss.
Let's make it this way: 
...
node.next = new Node<K, V>(key, value);
...
table[getIndex(key)] = new Node<K, V>(key,value);